frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/index.tsx

```diff
@@ -316,49 +282,60 @@ const UpdateStrategy: React.FC<SpecCapabilityProps> = ({
-    return <K8sResourceLinkCapability capability={capability} {...props} />;
-  }
+  const Component = React.useMemo(() => {
+    if (capability?.startsWith(SpecCapability.k8sResourcePrefix)) {
+      return K8sResourceLinkCapability;
+    }
 
-  if (capability?.startsWith(SpecCapability.selector)) {
-    return <BasicSelector capability={capability} {...props} />;
-  }
+    if (capability?.startsWith(SpecCapability.selector)) {
+      return BasicSelector;
+    }
 
-  switch (capability) {
-    case SpecCapability.podCount:
-      return <PodCount {...props} />;
-    case SpecCapability.endpointList:
-      return <Endpoints {...props} />;
-    case SpecCapability.label:
-      return <Label {...props} />;
-    case SpecCapability.namespaceSelector:
-      return <NamespaceSelector {...props} />;
-    case SpecCapability.resourceRequirements:
-      return <ResourceRequirements {...props} />;
-    case SpecCapability.booleanSwitch:
-      return <BooleanSwitch {...props} />;
-    case SpecCapability.password:
-      return <Secret {...props} />;
-    case SpecCapability.updateStrategy:
-      return <UpdateStrategy {...props} />;
-    case SpecCapability.checkbox:
-      return <CheckboxUIComponent {...props} />;
-    case SpecCapability.hidden:
-      return null;
-    default:
-      if (_.isObject(props.value)) {
-        // eslint-disable-next-line no-console
-        console.warn(
-          `[Invalid SpecDescriptor] Descriptor is incompatible with non-primitive value.`,
-          props.descriptor,
-        );
-      }
-      return <DefaultCapability {...props} />;
-  }
+    switch (capability) {
+      case SpecCapability.podCount:
+        return PodCount;
+      case SpecCapability.endpointList:
+        return Endpoints;
+      case SpecCapability.label:
+        return Label;
+      case SpecCapability.namespaceSelector:
+        return NamespaceSelector;
+      case SpecCapability.resourceRequirements:
+        return ResourceRequirements;
+      case SpecCapability.booleanSwitch:
+        return BooleanSwitch;
+      case SpecCapability.password:
+        return SecretCapability;
+      case SpecCapability.updateStrategy:
+        return UpdateStrategy;
+      case SpecCapability.checkbox:
+        return CheckboxUIComponent;
+      case SpecCapability.hidden:
+        return null;
+      default:
+        if (_.isObject(props.value)) {
+          // eslint-disable-next-line no-console
+          console.warn(
+            `[Invalid SpecDescriptor] Cannot render 'spec.${props.descriptor.path}'. A valid x-descriptor must be provided for non-primitive properties.`,
+            'See https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md#olm-descriptor-reference',
+          );
+          return null;
+        }
+        return DefaultCapability;
+    }
+  }, [props.descriptor, props.value, capability]);
+  return !Component ? null : (
```

Comment:
Not a big deal, but you could write this as 
```suggestion
  return Component && (
```